src/liballoc/collections/btree/node.rs

```diff
@@ -510,7 +513,7 @@ impl<'a, K, V, Type> NodeRef<marker::Mut<'a>, K, V, Type> {
     /// This also implies you can invoke this member on the shared root, but the resulting pointer
     /// might not be properly aligned and definitely would not allow accessing keys and values.
     fn as_leaf_mut(&mut self) -> *mut LeafNode<K, V> {
-        self.node.as_ptr()
+        unsafe { &mut *(self.node.as_ptr() as *mut LeafNode<K, V>) }
```

Comment:
Creating a reference here risks UB and is also completely pointless. Just use `self.node.as_ptr() as *mut LeafNode<K, V>`.

Reply:
Done, and alloc tests and Miri are happy